src/main.c

```diff
@@ -369,34 +367,23 @@ int main(int argc, char *argv[])
   }
 
   if (argc > 1 && argv[1][0] != '-') {
-    strncpy(sysout_name, argv[1], MAXPATHLEN);
+    strncpy(sysout_name_first_arg, argv[1], MAXPATHLEN);
     i++;
-  } else if ((envname = getenv("LDESRCESYSOUT")) != NULL) {
-    strncpy(sysout_name, envname, MAXPATHLEN);
-  } else if ((envname = getenv("LDESOURCESYSOUT")) != NULL)
-    strncpy(sysout_name, envname, MAXPATHLEN);
-  else {
-#ifdef DOS
-    strncpy(sysout_name, "lisp.vm", MAXPATHLEN);
-#else
-    if ((envname = getenv("HOME")) != NULL) {
-      strncpy(sysout_name, envname, MAXPATHLEN);
-      strncat(sysout_name, "/lisp.virtualmem", MAXPATHLEN - 17);
-    }
-#endif /* DOS */
   }
-  if (access(sysout_name, R_OK)) {
-    perror("Couldn't find a sysout to run");
-    (void)fprintf(stderr, "%s%s", helpstring, nethubHelpstring);
-    exit(1);
-  }
-  /* OK, sysout name is now in sysout_name, and i is moved past a supplied name */
+
 
   for (; i < argc; i += 1) { /* step by 1 in case of typo */
 
+   /* Check for -sysout arg */
```

Comment:
This could have a comment noting that it will not happen in the X11 case, but is handling -sysout for SDL or any other display subsystem that doesn't parse its own command line for -sysout.

Reply:
That is true for many (most?) of the arguments being processed in this argument processing loop - they have already been processed and removed from argv by the call to read_Xoption.  They are "repeated" in this argument processing loop for the non-X platforms.  I added a comment at the top of the loop to note that this is the case.